Групповой проект

geofij opened the pull request.

Commit:
feat: Добавил каскадное удаление для существующих таблиц
Добавил новые таблицы для реализации ТЗ:
- review
- review_like
- feed_operation
- feed_type
- feed
- director
- film_director

Commit:
feat: Добавил заполнение таблиц feed_type, feed_operation

Commit:
feat: Добавил схему БД в readme.md

Commit:
feat: Удалил схему БД в readme.md

Commit:
feat: Удалил схему БД в readme.md

Commit:
feat: Добавил удаление фильмов и пользователей

Commit:
fix: Исправил переменную пути в удалении фильма

Commit:
fix: Если фильм/юзер не найден - кидаю исключение

Commit:
fix: Если фильм/юзер не найден - --кидаю исключение-- исключение выкинетс] само ))

radya100y committed:
Merge pull request #8 from geofij/add-remove-endpoint

Add remove endpoint

Commit:
fix: Внес изменения в структуру БД - таблицы отзывов и лайков отзывов

Commit:
feat: Добавил модель отзывов

Commit:
feat: Добавил модель отзывов

Commit:
feat: Добавил интерфейс хранения отзывов

Commit:
feat: Добавил базовые штуки в сторадж

Commit:
fix: отвязал класс Review от базового суперкласса

Commit:
feat: Добавил класс сервисов на отзывы

Commit:
feat: Добавил реакции на отзывы

Commit:
feat: Добавил в контроллер базовые вызовы

Commit:
feat: Добавил в контроллер вызов всех/по фильму отзывов

Commit:
feat: Добавил в контроллер реакции на отзывы

Commit:
fix: checkstyle

sergeyk13 committed:
изменил схему
изиенил дата файл
реализовал логику feed

sergeyk13 committed:
изменил схему
изиенил дата файл
реализовал логику feed

radya100y committed:
Merge pull request #9 from geofij/add-reviews

Add reviews

sergeyk13 committed:
Merge branch 'develop' into add-feed

VitalyPotapenko committed:
добавлена функциональность "Рекомендации"

VitalyPotapenko committed:
починка №1 упавшей checkstyle проверки при создании пуллреквеста

sergeyk13 committed:
коммит 1.0

geofij committed:
feat: функциональность режиссер. refactor: в схеме данных удаление всех таблиц. В маппере фильма при дополнительных запросах подстановка film_id, а не id.

geofij committed:
style: правка отступа.

geofij committed:
fix: тест хранилища фильмов.

VitalyPotapenko committed:
добавляю функциональность "Вывод самых популярных фильмов по жанру и годам"

VitalyPotapenko committed:
правки после ревью

VitalyPotapenko committed:
Merge pull request #10 from geofij/add-recommendations

добавлена функциональность "Рекомендации"

geofij committed:
feat: логирование при удалении режиссера, проверка на существование режиссера при обновлении. refactor: убрала лишнее ветвление при обновлении жанров и режиссеров фильма. Возвращение фильма, директора и юзера при их обновлении.

geofij committed:
Merge pull request #11 from geofij/add-director

Функциональность "Режиссер"

sergeyk13 committed:
Merge branch 'develop' into add-feed

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java
#	src/main/resources/schema.sql

sergeyk13 committed:
пуш для Родиона

sergeyk13 committed:
коммит 1.0.0

sergeyk13 committed:
коммит 1.1.0
подправил схему

sergeyk13 committed:
коммит 1.1.1
исправил log в FeedStorage

sergeyk13 committed:
коммит 1.2.1
исправил для прохождения теста

VitalyPotapenko committed:
checkstyle fix#1 for add-most-populars

4lovi4 committed:
Add search (#14)

* feat: Добавлен поиск по названию фильма

* fix: добавил сортировку по размеру множества режиссеров для найденных фильмов иначе тест не проходит

* fix: checkstyle перенёс из начала строки {

* fix: checkstyle перенёс else if

* fix: checkstyle перенос строки перед закрывающей }

* fix: checkstyle Перенос перед } и комментарий к пустому методу

* fix: использовал подзапрос для поиска по подстроке имени режиссёра вместо нескольких join

---------

Co-authored-by: a.a.dyatlov <a.a.dyatlov@tinkoff.ru>

VitalyPotapenko committed:
fixes for remarks on add-most-populars review by AlexeyD

VitalyPotapenko committed:
Merge branch 'develop' into add-most-populars

VitalyPotapenko committed:
Merge pull request #17 from geofij/add-most-populars

добавляю функциональность "Вывод самых популярных фильмов по жанру и годам"

radya100y committed:
Merge branch 'develop' into add-feed

radya100y committed:
Merge pull request #16 from geofij/add-feed

Add feed

sergeyk13 committed:
Add common films (#15)

* коммит 1.0

* коммит 1.0
в UserStorage добавил проверку на дружбу
в FilmService, FilmStorage вывод общих друзей

* коммит 1.1
в FilmService кинул в коммменты проверку на дружбу

* коммит 1.2
в FilmService кинул в коммменты проверку на дружбу

* коммит 1.2.1
в FilmService кинул в коммменты проверку на дружбу

* коммит 1.3.0
в UserStorage убрал friendStorage

* коммит 1.2.3
подправил ревью убрал код коментах

* fix code-style and missed import statement for add-comon-films

---------

Co-authored-by: VitalyPotapenko <121198959+VitalyPotapenko@users.noreply.github.com>
Co-authored-by: VitalyPotapenko <vitaly.a.potapenko@yandex.ru>

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -4,36 +4,49 @@
+    private final FeedStorage feedStorage;
+    private final DirectorStorage directorStorage;
 
     @Autowired
```
Аннотация не обязательна над конструктором, если он один

VitalyPotapenko commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -2,21 +2,25 @@
 public class UserController {
     private final UserService service;
+    private final FeedService feedService;
     private long idCounter;
```
Александр, привет. Я Виталий - в рамках этого группового проекта в нашей команде тимлид Марина. Я работаю с замечаниями по ревью, поэтому отвечу в это обсуждение.

Короткий ответ: "Да, этот сервис здесь нужен". Сервис FeefService реализует функционал "Лента событий". Согласно ТЗ (см приложенный к этому сообщению скриншот ТЗ "ТЗ_Лента_Событий.png") ряд действий пользователя должен попадать в ленту событий (как то - добавление в друзья, удаление из друзей) должно попадать в ленту событий пользователя. И именно с этой целью в класс UserController был добавлен FeedService 
![ТЗ_Лента_Событий](https://github.com/geofij/java-filmorate/assets/121198959/194717f0-83b7-4e5a-869f-6c60217744e6)

Alexander-Bolotov replied:
Привет!
Видимо, я не туда тыкнул для комментария)
Я имел ввиду про idCounter;

VitalyPotapenko replied:
понял) Такая схема работы с id для фильмов приехала к нам при клонировании проекта, сами мы его не трогали в рамках работы над проектом. Но исправили, и в пулл реквесте, который сейчас проходит внутрикомандное ревью, поле idCounter убрано)

VitalyPotapenko committed:
Add common films (#20)

* коммит 1.0

* коммит 1.0
в UserStorage добавил проверку на дружбу
в FilmService, FilmStorage вывод общих друзей

* коммит 1.1
в FilmService кинул в коммменты проверку на дружбу

* коммит 1.2
в FilmService кинул в коммменты проверку на дружбу

* коммит 1.2.1
в FilmService кинул в коммменты проверку на дружбу

* коммит 1.3.0
в UserStorage убрал friendStorage

* коммит 1.2.3
подправил ревью убрал код коментах

* fix code-style and missed import statement for add-comon-films

* коммит 1.0
в UserStorage добавил проверку на дружбу
в FilmService, FilmStorage вывод общих друзей

* коммит 1.3.0
в UserStorage убрал friendStorage

* fix review remarks

* fix genres set order in return statement of create method

---------

Co-authored-by: Sergey <45943782+sergeyk13@users.noreply.github.com>

VitalyPotapenko commented on src/main/java/ru/yandex/practicum/filmorate/controller/DirectorController.java:
```diff
@@ -0,0 +1,49 @@
+
+    @GetMapping
+    public List<Director> getDirectors() {
+        log.info("Getting all directors");
```
поправили

VitalyPotapenko commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -35,9 +52,8 @@ public Film create(@Valid @RequestBody Film film) {
     @PutMapping
```
починили

VitalyPotapenko commented on src/main/java/ru/yandex/practicum/filmorate/storage/db/FeedDbStorage.java:
```diff
@@ -0,0 +1,72 @@
+import java.util.LinkedHashSet;
+import java.util.List;
+
+@Component
```
сделали @Repository и тут тоже
